components/create_user_groups_modal/create_user_groups_modal.tsx

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {Modal} from 'react-bootstrap';
+
+import {FormattedMessage} from 'react-intl';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import * as Utils from 'utils/utils.jsx';
+import {GroupCreateWithUserIds} from 'mattermost-redux/types/groups';
+
+import 'components/user_groups_modal/user_groups_modal.scss';
+import './create_user_groups_modal.scss';
+import {ModalData} from 'types/actions';
+import Input from 'components/input';
+import AddUserToGroupMultiSelect from 'components/add_user_to_group_multiselect';
+import {ActionResult} from 'mattermost-redux/types/actions';
+import LocalizedIcon from 'components/localized_icon';
+import {t} from 'utils/i18n';
+
+export type Props = {
+    onExited: () => void;
+    backButtonCallback?: () => void;
+    actions: {
+        createGroupWithUserIds: (group: GroupCreateWithUserIds) => Promise<ActionResult>;
+        openModal: <P>(modalData: ModalData<P>) => void;
+    };
+}
+
+type State = {
+    show: boolean;
+    name: string;
+    mention: string;
+    savingEnabled: boolean;
+    usersToAdd: UserProfile[];
+    mentionUpdatedManually: boolean;
+    mentionInputErrorText: string;
+    nameInputErrorText: string;
+    showUnknownError: boolean;
+}
+
+export default class CreateUserGroupsModal extends React.PureComponent<Props, State> {
```

Comment:
nit: Again, not an stopper, but we could subdivide this on smaller functional components?